library/ecp.c

```diff
@@ -2526,7 +2526,7 @@ static int ecp_mul_mxz( mbedtls_ecp_group *grp, mbedtls_ecp_point *R,
     MBEDTLS_MPI_CHK( ecp_randomize_mxz( grp, &RP, f_rng, p_rng ) );
 
     /* Loop invariant: R = result so far, RP = R + P */
-    i = mbedtls_mpi_bitlen( m ); /* one past the (zero-based) most significant bit */
+    i = grp->pbits; /* one past the (zero-based) most significant bit */
```

Comment:
While this works in practice, I think using `grp->nbits + 1` would be more correct as a matter of principle. `nbits` is what's used when generating private keys in `mbedtls_ecp_gen_privkey()` and when checking them in `mbedtls_ecp_check_privkey()`, and this is consistent with the documentation of the field in `struct mbedtls_ecp_group`.

(Of course, there are only two Montgomery curve in the code base, they both have `nbits + 1 == pbits` and that's not a coincidence. Still, while we're making the code more obviously correct, let's go all the way.)

Reply:
(minor: presumably the comment is no longer valid?)